src/coinbase/smart_wallet.ts

```diff
@@ -0,0 +1,110 @@
+import { NetworkIdentifier, SmartWallet as SmartWalletModel } from "../client";
+
+import { Coinbase } from "./coinbase";
+
+import type { Hex } from 'viem';
+import type {
+  UserOperationCalls
+} from 'viem/account-abstraction';
+
+import { encodeFunctionData, LocalAccount } from "viem";
+import { UserOperation } from "./user_operation";
+
+export class SmartWallet {
+  private model: SmartWalletModel;
+  private account: LocalAccount;
+  private networkId?: NetworkIdentifier;
+
+  public constructor(model: SmartWalletModel, account: LocalAccount) {
+    this.model = model;
+    this.account = account;
+  }
+
+  public static async create({account}: {account: LocalAccount}): Promise<SmartWallet> {
+    const result = await Coinbase.apiClients.smartWallet!.createSmartWallet({
+        owner: account.address,
+    });
+
+    const smartWallet = new SmartWallet(result.data!, account);
+    return smartWallet;
+  }
+
+  public static async connect({smartWalletAddress, account}: {smartWalletAddress: string, account: LocalAccount}): Promise<SmartWallet> {
```

Comment:
It's an interesting experience reading these APIs without documentation - because without the JSDoc, I can't immediately grok what they would do.

I understand there are schelling points with Gnosis Safe, but I personally haven't used it, so I don't have a point of comparison.

Seems to me that, in a vacuum, this method should be called `fetch`. Just my 2c

Reply:
imo connect makes sense, you're not really fetching anything (well you are literally calling an api...), the job to be done here is get a SmartWallet object from the users perspective, fetching is an implementation detail. Would vote `connect` for convention